src/posix/platform/system.cpp

```diff
@@ -63,6 +63,8 @@ otInstance *otSysInit(otPlatformConfig *aPlatformConfig)
     instance = otInstanceInitSingle();
     assert(instance != nullptr);
 
+    platformBackboneInit(instance, aPlatformConfig->mBackboneInterfaceName);
```

Comment:
add `OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE` condition?

Reply:
It's actually not a must. 
- It's not a strong requirement of posix platform code to reduce the code size, so we can have this code running even when it's not a Backbone Router. 
- Further more, it's possible (and perfectly fine) that other COAP objects (other than Backbone TMF agent) might need to bind on the Backbone interface.

I don't have a strong opinion for it, but I also think it's fine to keep the code available on non Backbone Routers. 

Thoughts? @librasungirl

Reply:
I may prefer to add the macro condition: 
- follow the same style as the others, like `platformNetifProcess()` has `OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE` wrapper
- As similar rule in another PR https://github.com/openthread/openthread/pull/5346#discussion_r475161377 agreed -better to only add required functionalities unless we have clear use case for the `possible (and perfectly fine) that other COAP objects (other than Backbone TMF agent) might need to bind on the Backbone interface`, though we don't concern code size on posix so much as in embedded platform.